h2o-algos/src/main/java/hex/glm/ComputationState.java

```diff
@@ -86,6 +110,33 @@ public ComputationState(Job job, GLMParameters parms, DataInfo dinfo, BetaConstr
     }
     _modelBetaInfo = bi;
   }
+  
+  // set _noReg to true if l1 and l2 regularization are both zero.
+  public void setNoReg() {
+    _noReg = (_lambda == 0.0);
```

Comment:
Maybe `&& _parms._lambda_search=false`?

I am currently not able to compile it. Otherwise I'd try to debug it more. My concern is `coldstart` which sets the lambda to 0.0 and `copyCheckModel2State` which updates the `_lambda` but not `_noReg`.

I think calling `setNoReg` in `setLambda` and `setLambdaSimple` would ensure consistency (but it might be completely unnecessary).

Reply:
When lambda_search = True and you set lambda_=0.  There will be no lambda search just lambda set at that one value.  I moved it to be inside linear constraint check as it makes more sense.  Thanks.